### PR TITLE
added `EduStat` to the `EduPlugin` containing info about interactive …

### PR DESCRIPTION
**Pull request by AlexanderPrendota**
…challenges, quizzes and lessons/sections

### CODE REVIEW

File: intellij-plugin-structure/structure-edu/src/main/kotlin/com/jetbrains/plugin/structure/edu/EduPlugin.kt [hunk 1]
```diff
@@ -27,4 +30,53 @@ data class EduPlugin(
...
+
+  fun countTheoryTasks(): Int {
+    return tasks[TaskType.THEORY.id] ?: 0
+  }
+
+  companion object {
+    fun fromDescriptor(descriptor: EduPluginDescriptor): EduStat {
+      val allItems = mutableListOf<EduItem>()
+      allItems += descriptor.items
+      descriptor.items.flatMapTo(allItems, { it.items })
```
Comment (AlexanderPrendota) [inline intellij-plugin-structure/structure-edu/src/main/kotlin/com/jetbrains/plugin/structure/edu/EduPlugin.kt:60]: why do you need to do this? 

why not just `descriptor.items.filter {} `

File: intellij-plugin-structure/structure-edu/src/main/kotlin/com/jetbrains/plugin/structure/edu/EduPlugin.kt [hunk 2]
```diff
@@ -19,12 +22,61 @@ data class EduPlugin(
...
\ No newline at end of file
+}
+
+data class EduStat(
+  val sections: List<String>,
+  val lessons: List<String>,
+  val tasks: Map<String, Int>
+) {
+
+  fun countInteractiveChallenges(): Int {
```
Comment (AlexanderPrendota) [inline intellij-plugin-structure/structure-edu/src/main/kotlin/com/jetbrains/plugin/structure/edu/EduPlugin.kt:41]: as I understood there are methods for tests. Let's put in test module

Comment (ktisha) [reply intellij-plugin-structure/structure-edu/src/main/kotlin/com/jetbrains/plugin/structure/edu/EduPlugin.kt:41]: I've added them not for tests but for the easy access to the information needed in the additional information section.
image (1)

File: intellij-plugin-structure/structure-edu/src/main/kotlin/com/jetbrains/plugin/structure/edu/EduPlugin.kt (hunk 1 above)
Comment (ktisha) [reply intellij-plugin-structure/structure-edu/src/main/kotlin/com/jetbrains/plugin/structure/edu/EduPlugin.kt:60]: There are 2 types of top-level items in the course -- sections and lessons. And sections, in turn, could contain lessons. Therefore I need to collect all the items before filtering.

File: intellij-plugin-structure/structure-edu/src/main/kotlin/com/jetbrains/plugin/structure/edu/EduPlugin.kt (hunk 2 above)
Comment (AlexanderPrendota) [reply intellij-plugin-structure/structure-edu/src/main/kotlin/com/jetbrains/plugin/structure/edu/EduPlugin.kt:41]: I will do it on the Marketplace side

File: intellij-plugin-structure/structure-edu/src/main/kotlin/com/jetbrains/plugin/structure/edu/EduPlugin.kt (hunk 1 above)
Comment (AlexanderPrendota) [reply intellij-plugin-structure/structure-edu/src/main/kotlin/com/jetbrains/plugin/structure/edu/EduPlugin.kt:60]: maybe `val all = descriptor.items.flatMap { it.items } + descriptor.items` ?

**Review by serejke — APPROVED** [review]

**Pull request merged**